File: src/leetCode/Ex61.java
package leetCode;

public class Ex61 {
    public ListNode rotateRight(ListNode head, int k) {
        if (head == null || head.next == null) {
            return head;
        }
        int len = 1;
        ListNode dump = new ListNode(0);
        dump.next=(head);
        //t1 is the tail
        while (head.next != null) {
            len++;

        }
        int kk = k % len;
        if (kk == 0) {
            return head;
        }
//        for (int i = 0; i < len - kk - 1; i++) {
//
//        }
        dump.next= head.next;
        head.next=(null);
        head.next=(head);
        return dump.next;
    }
}

File: src/leetCode/ex61.kt
package leetCode


fun rotateRight(head: ListNode?, k: Int): ListNode? {
    if (head?.next == null) return head
    val dump = ListNode()
    dump.next = head

    var t1 = head
    var t2 = head
    var len = 1
    while (t1?.next != null) {
        t1 = t1.next
        len++
    }
    val kk = k % len
    for (i in 0 until len - kk - 1) {
        t2 = t2?.next
    }
    dump.next = t2?.next
    t2?.next = null
    t1?.next = head
    return dump.next
}


fun main(args: Array<String>) {
    val start = ListNode(1)
    start.next = ListNode(2)
    start.next?.next = ListNode(3)
    rotateRight(start, 1).apply {
        var t = this
        while (t != null) {
            println(t.`val`)
            t = t.next
        }
    }
}